src/draw/external_image.rs

```diff
@@ -0,0 +1,56 @@
+use webrender::api::*;
+
+use render::RenderBuilder;
+use widget::draw::Draw;
+use resources::resources;
+use geometry::{Rect, RectExt, Size, SizeExt};
+
+pub struct ExternalImageState {
```

Comment:
Do you think it makes sense to change the name of this to TextureState? I think ExternalImage is more a webrender implementation detail than something that would be familiar to an end user. I also considered ImageBuffer and Canvas, but I think maybe ImageBuffer is not specific enough, and Canvas could be reserved for vector graphics.

Reply:
Yeah… I was thinking that "texture" might be confusing, as in, just using textures in the UI, like, not in the OpenGL sense but in the "image pattern" sense.

GLCanvas?

Reply:
Though I guess `Texture` is fine with documentation…

Reply:
Yeah I like GLCanvas a lot, GLTexture could be good too?